ovh/resource_domain_name_servers.go

```diff
@@ -0,0 +1,220 @@
+package ovh
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/go-ovh/ovh"
+	"log"
+	"strings"
+)
+
+type OvhDomainNameServerUpdate struct {
+	NameServers []OvhDomainNameServer `json:"nameServers,omitempty"`
+}
+
+type OvhDomainNameServerUpdateResult struct {
+	CanAccelerate bool   `json:"canAccelerate,omitempty"`
+	CanCancel     bool   `json:"canCancel,omitempty"`
+	CanRelaunch   bool   `json:"canRelaunch,omitempty"`
+	Comment       string `json:"comment,omitempty"`
+	CreationDate  string `json:"creationDate,omitempty"`
+	Domain        string `json:"domain,omitempty"`
+	DoneDate      string `json:"doneDate,omitempty"`
+	Function      string `json:"function,omitempty"`
+	Id            int64  `json:"id,omitempty"`
+	LastUpdate    string `json:"lastUpdate,omitempty"`
+	Status        string `json:"status,omitempty"`
+	TodoDate      string `json:"todoDate,omitempty"`
+}
+
+type OvhDomainNameServer struct {
+	Id       int64  `json:"id,omitempty"`
+	Host     string `json:"host,omitempty"`
+	Ip       string `json:"ip,omitempty"`
+	IsUsed   bool   `json:"isUsed,omitempty"`
+	ToDelete bool   `json:"toDelete,omitempty"`
+}
+
+type OvhDomainNameServers struct {
+	Id      string                `json:"id,omitempty"`
+	Domain  string                `json:"domain,omitempty"`
+	Servers []OvhDomainNameServer `json:"servers,omitempty"`
+}
+
+func (r *OvhDomainNameServers) String() string {
+	domains := make([]string, 0)
+	for _, server := range r.Servers {
+		domains = append(domains, fmt.Sprintf(
+			"server[id: %v, host: %s, ip: %s, isUsed: %v, toDelete: %v]",
+			server.Id,
+			server.Host,
+			server.Ip,
+			server.IsUsed,
+			server.ToDelete,
+		))
+	}
+	return fmt.Sprintf(
+		"nameservers[id: %v, domain: %s, servers: [%s]]",
+		r.Id,
+		r.Domain,
+		strings.Join(domains, ", "),
+	)
+}
+
+func resourceOvhDomainNameServersImportState(
+	d *schema.ResourceData,
+	meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	d.SetId(givenId)
+	d.Set("domain", d.Id())
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func resourceOvhDomainNameServers() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOvhDomainNameServersCreate,
+		Read:   resourceOvhDomainNameServersRead,
+		Update: resourceOvhDomainNameServersCreate, // Update is the same as create
+		Delete: resourceOvhDomainNameServersDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceOvhDomainNameServersImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"servers": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host": {
+							Type:     schema.TypeString,
+							Required: true,
```

Comment:
I guess that this API require either `host` either `ip`, but `host` can be optional if `ip` is provided.

Reply:
API requires a string value for host from specs

![image](https://github.com/ovh/terraform-provider-ovh/assets/111863168/18325e3b-d35e-45d0-9fd1-f542270d8138)

https://eu.api.ovh.com/console/?section=%2Fdomain&branch=v1#post-/domain/-serviceName-/nameServers/update

Reply:
I think what @rbeuque74 said is still valid, neither host nor ip are required

Reply:
After clarification internally in OVHcloud, you are right, that's the requirements that is enforce on the API.
Sorry about misunderstanding